pyEpiabm/pyEpiabm/intervention/testing.py

```diff
@@ -0,0 +1,68 @@
+#
+# Gives people a positive or negatie test status
+#
+import random
+
+from pyEpiabm.property import InfectionStatus
+from pyEpiabm.intervention import AbstractIntervention
+
+
+class Testing(AbstractIntervention):
+    """ Class to move through testing queue and assign
+    positive test results depending on true/false positive rates.
+
+    """
+
+    def __init__(self,
+                 testing_capacity,
+                 false_positive,
+                 false_negative,
+                 population,
+                 **kwargs
+                 ):
+        self.testing_capacity = testing_capacity
+        self.false_positive = false_positive
+        self.false_negative = false_negative
+
+        population.test_count = [0, 0]
+
+        super(Testing, self).__init__(population=population, **kwargs)
+
+    def __call__(self, time):
+        for cell in self._population.cells:
+            num_pcr = 0
+            num_lft = 0
+            while (num_pcr < self.testing_capacity[0] and not
+                   cell.PCR_queue.empty()):
+                r = random.random()
+                person = cell.PCR_queue.get()
+                num_pcr += 1
+                if (person.is_symptomatic() or
+                   person.infection_status == InfectionStatus.InfectASympt):
+                    if r > self.false_negative[0]:
+                        person.date_positive = time
+                        self._population.test_count[0] += 1
+                else:
+                    if r < self.false_positive[0]:
```

Comment:
I'm not sure about this probability - please review. I can go through the maths if you're convinced it is right (as I may be wrong) though

Reply:
I'm pretty sure I've got it right - we could maybe look over it tomorrow.

Reply:
Okay I've had another look at that was completely my mistake - you are right!